File: Greedy/0984.String Without AAA or BBB/Solution.java
class Solution {
    public String strWithout3a3b(int a, int b) {
        String ans = "";
        while (a > 0 || b > 0) {
            if (a >= b) {
                if (ans.length() >= 2 && ans.substring(ans.length() - 2).equals("aa")) {
                    ans += "b";
                    b--;
                } else {
                    ans += "a";
                    a--;
                }
            } else {
                if (ans.length() >= 2 && ans.substring(ans.length() - 2).equals("bb")) {
                    ans += "a";
                    a--;
                } else {
                    ans += "b";
                    b--;
                }
            }
        }
        return ans;
    }
}

File: Greedy/0984.String Without AAA or BBB/Solution.py
class Solution:
    def strWithout3a3b(self, a: int, b: int) -> str:
        ans = ""
        while a or b:
            if a >= b:
                if len(ans) >= 2 and ans[-2:] == 'aa':
                    ans += 'b'
                    b -= 1
                else:
                    ans += 'a'
                    a -= 1
            else:
                if len(ans) >= 2 and ans[-2:] == 'bb':
                    ans += 'a'
                    a -= 1
                else:
                    ans += 'b'
                    b -= 1
        return ans